exercises/anagram/canonical-data.json

```diff
@@ -109,13 +118,13 @@
       "expected": []
     },
     {
-      "description": "words are not anagrams of themselves (case-insensitive)",
+      "description": "words are not anagrams of themselves (case-insensitive), mixed",
```

Comment:
Since the test description has changed, perhaps this is a new test, not a replacement test.

Reply:
There should be a test which excludes all possible capitalizations of the input string and matches at least one valid anagram at the same time - adding another test seemed a bit  repetetive, as the changed test already performed the first part.

Reply:
But there shouldn't be a test that covers the candidates that are mixed case of the original word and expects no match?

Reply:
The modified should perform the same function, shouldn't it? Or do you think that just the same word with different capitalizations could be an edge case in someone's code and should be tested for separately?

Reply:
I am saying, basically, that this test was changed in regards to the expected as well, which indicates the the test that is being replaced was unnecessary, or testing the wrong thing.

I think adding the test and later deciding that the test concerning "bananas" is not needed is a better story line.

Reply:
It feels like a new test, rather than a modification of the test.

Reply:
Fair enough, I guess I'll create another test and only try checking for two case variants of the input.

Should I force-push this over the previous commit?

Reply:
Feel free to force push on your branches for the pull requests, as it helps keep things clean.

Reply:
It doesn't really matter. We'll squash merge. Do what you'd like; keeping the changes clean is helpful but in this case probably just fine anyway :)

Reply:
@SleeplessByte I don't always squash merge, it depends on how the history presented in the commits.  I didn't now that it was unconditional that we would do that.

Reply:
I also don't _always_ squash merge, but I think for about 95% of all PRs across most exercism repositories, squashing make sense ;)

In this case: adding a set of test cases in a single file usually doesn't really benefit from having more "history" than just a single commit, agreed?

Reply:
Yeah, which I was thinking they could force push.  I thought that was why the force push might be asked about, otherwise it would probably be a normal push, and I encourage changing the history locally, so we don't have to remember to need to here remotely.

Reply:
Got ya! I'm striving to have people squash merge, for the simple reason that it also adds a link back to this PR!

Reply:
That's a nice feature.  I didn't know that happens.